armory/datasets/preprocessing.py

```diff
@@ -0,0 +1,101 @@
+"""
+Standard preprocessing for different datasets
+"""
+
+
+import tensorflow as tf
+
+
+REGISTERED_PREPROCESSORS = {}
+
+
+def register(function, name=None):
+    if name is None:
+        name = function.__name__
+    global REGISTERED_PREPROCESSORS
+    REGISTERED_PREPROCESSORS[name] = function
+    return function
+
+
+def list_registered():
+    return list(REGISTERED_PREPROCESSORS)
+
+
+def get(name, version=None):
+    if version is not None:
+        raise NotImplementedError("version is not None for preprocessing to_canon")
+    if name not in REGISTERED_PREPROCESSORS:
+        raise KeyError(
+            f"prepreprocessor {name} not registered. Use one of {list_registered()}"
```

Comment:
based on how things are currently set up, `name` will always simply be the name of the dataset. So let's say you're using a new (to Armory), say, image classification dataset e.g. `fashion_mnist`. I don't see an easy way to use your new dataset with an _existing_ preprocessing function, e.g. mnist's. A lot of these preprocessing functions will look the same (`(image_to_canon(element["image"]), element["label"])`) for different datasets, so it'd be a bit cumbersome to need to implement a new function:
```
def fashion_mnist(element):
    return (image_to_canon(element["image"]), element["label"])
```
that's a replica of existing ones.

Reply:
That's a good call. I made an easy way to reference them.